garnett/trajectory.py

```diff
@@ -756,18 +756,18 @@ def arrays_loaded(self):
         """Returns true if arrays are loaded into memory.
 
         See also: :meth:`~.load_arrays`"""
-        return not (self._N is None or
-                    self._type is None or
-                    self._types is None or
-                    self._type_ids is None or
-                    self._positions is None or
-                    self._orientations is None or
-                    self._velocities is None or
-                    self._mass is None or
-                    self._charge is None or
-                    self._diameter is None or
-                    self._moment_inertia is None or
-                    self._angmom is None or
+        return not (self._N is None and
+                    self._type is None and
+                    self._types is None and
+                    self._type_ids is None and
+                    self._positions is None and
+                    self._orientations is None and
+                    self._velocities is None and
+                    self._mass is None and
+                    self._charge is None and
+                    self._diameter is None and
+                    self._moment_inertia is None and
+                    self._angmom is None and
                     self._image is None)
```

Comment:
The behavior of this function had to be changed in order to enable raise `AttributeError` for `None` properties. However, I'm not 100% convinced this is the right approach. Thoughts?

Reply:
Could you slightly elaborate on the problem?

Reply:
Sure. The purpose of this PR is to implement raising `AttributeError` whenever invalid (`None`) trajectory attributes are accessed, analogous to the `Frame` objects. 

Previously, the `arrays_loaded` function (which is called within the `_assertarrays_loaded` every time a property is accessed) returned `False` if **any** of the listed attributes was `None`.  Because of this, the `RuntimeError` below was always triggered before the `AttributeError`.

```python
def _assertarrays_loaded(self):
    "Raises a RuntimeError if trajectory arrays are not loaded."
    if not self.arrays_loaded():
        raise RuntimeError(
            "Trajectory arrays not loaded! Use load_arrays() or load().")
```

Thus, to me it made more sense at the moment that the `arrays_loaded` function returns `False` if **all** the attributes are `None`.

Reply:
So the basic assumption is that the arrays are loaded if *at least one* attribute is not `None`, correct?

In this case the logic here is correct. It might be a little bit more obvious if expressed like this:
```
return any(
    self._N is not None or
    self._type is not None or
    ...
)
```

I would also suggest to consider:
```
class Trajectory:
...
    TRAJ_ATTRIBUTES = ['N', 'type', 'types', ... ]

    def arrays_loaded(self):
        return any(getattr(self, '_' + key) is not None for key in self.TRAJ_ATTRIBUTES)
```
I'm not sure if that's better, but having that list as part of the class definition might also be useful in other cases.

Reply:
I agree with both of @csadorf's recommendations -- the `TRAJ_ATTRIBUTES` in particular would help simplify code in several places. There's something similar here: https://github.com/glotzerlab/garnett/blob/e2efd562494d8ae82f9bf2eb0f35c85bb99649ec/garnett/trajectory.py#L840-L843

Reply:
> So the basic assumption is that the arrays are loaded if _at least one_ attribute is not `None`, correct?

 Yes, thats exactly what I meant.

> In this case the logic here is correct. It might be a little bit more obvious if expressed like this:
> 
> ```
> return any(
>     self._N is not None or
>     self._type is not None or
>     ...
> )
> ```
> 
> I would also suggest to consider:
> 
> ```
> class Trajectory:
> ...
>     TRAJ_ATTRIBUTES = ['N', 'type', 'types', ... ]
> 
>     def arrays_loaded(self):
>         return any(getattr(self, '_' + key) is not None for key in self.TRAJ_ATTRIBUTES)
> ```
> 
> I'm not sure if that's better, but having that list as part of the class definition might also be useful in other cases.

Thanks that was helpful